pkg/controller/storageprofile-controller.go

```diff
@@ -78,11 +92,7 @@ func (r *StorageProfileReconciler) Reconcile(_ context.Context, req reconcile.Re
 		return reconcile.Result{}, r.deleteStorageProfile(req.NamespacedName.Name, log)
 	}
 
-	if _, err := r.reconcileStorageProfile(storageClass); err != nil {
-		return reconcile.Result{}, err
-	}
-
-	return reconcile.Result{}, r.computeMetrics()
+	return r.reconcileStorageProfile(storageClass)
```

Comment:
We update the metric of a specific SC only if no error occured, which seems to be a consistent behavior. In case of SC deletion, we delete the metric.